src/libmongoc/tests/test-mongoc-sdam-monitoring.c

```diff
@@ -825,19 +828,16 @@ _test_heartbeat_fails_dns (bool pooled)
       client_set_heartbeat_event_callbacks (client, &context);
    }
 
-   start = bson_get_monotonic_time ();
-
    /* trigger "ismaster" handshake */
    r = mongoc_client_command_simple (
       client, "admin", tmp_bson ("{'foo': 1}"), NULL, NULL, &error);
 
+   /* This should result in either a DNS failure or connection failure depending on the network. */
```

Comment:
This is not due to new behavior. doesntexist.foobar may result in an immediate DNS failure, but getaddrinfo may also hang until server selection times out. I observed this on my local network. I don't think the tests should be relying on getaddrinfo returning immediately on a bad address if it is network dependent.